### ISSUE TITLE
Interlocked.Read for int?

### ISSUE THREAD

**Issue by nopara73** [nopara73/MagicalCryptoWallet#78]
Whenever I'm using Interlocked, I always end up using it with a long, due to it's the only type that `Interlocked.Read()` is implemented for.

Is it ok to write an extension method for int, too?

**Comment by lontivero** [comment]
`Interlocked.Read` is only available for `long` because it is only useful to ensure an atomic read of a 64 bits value under a 32 bits system.

[MSDN documentation](https://msdn.microsoft.com/en-us/library/system.threading.interlocked.read(v=vs.110).aspx) says:

>The Read method is unnecessary on 64-bit systems, because 64-bit read operations are already atomic. On 32-bit systems, 64-bit read operations are not atomic unless performed using Read. 

[This geekswithblog blog post](http://geekswithblogs.net/BlackRabbitCoder/archive/2012/08/23/c.net-little-wonders-interlocked-read-and-exchange.aspx) says the same: 

>When you are dealing with an item smaller or equal to the system word size (such as an int on a 32 bit system or a long on a 64 bit system) then the read is generally atomic, because it can grab all of the bits needed at once.  However, when dealing with something larger than the system word size (reading a long on a 32 bit system for example), it cannot grab the whole value at once, which can lead to some problems since this read isn’t atomic.

In summary, 32bits read operations are always atomic in both 32 bits and 64 bits systems so, Interlocked.Read(Int32) is unnecesary.

**Comment by nopara73** [comment]
Fair enough. Although I never used Interlocked for atomicity, always for thread safety.
